patches/0018-LPD-33712-Add-css-to-combopanel-when-maximized.patch

```diff
@@ -0,0 +1,30 @@
+From 985aa1817817ae82b705507c5c437ffad31106ca Mon Sep 17 00:00:00 2001
+From: Fortunato <fortunato.maldonado@liferay.com>
+Date: Thu, 15 Aug 2024 08:54:10 -0600
+Subject: [PATCH] LPD-33712 Add css to combopanel when maximized
+
+---
+ plugins/stylescombo/plugin.js | 10 ++++++++++
+ 1 file changed, 10 insertions(+)
+
+diff --git a/plugins/stylescombo/plugin.js b/plugins/stylescombo/plugin.js
+index 8b2b24d681..1818263c8d 100644
+--- a/plugins/stylescombo/plugin.js
++++ b/plugins/stylescombo/plugin.js
+@@ -184,6 +184,16 @@
+ 
+ 					if ( !counter[ CKEDITOR.STYLE_OBJECT ] )
+ 						this.hideGroup( lang[ 'panelTitle' + String( CKEDITOR.STYLE_OBJECT ) ] );
++
++					var combopanel = document.querySelector('.cke_panel.cke_combopanel');
++
++					var maximizedEditor = document.querySelector('.cke_inner.cke_maximized');
++
++					if (combopanel && maximizedEditor) {
++						combopanel.classList.add('cke_maximized');
++					} else {
++						combopanel.classList.remove('cke_maximized');
++					}
```

Comment:
Can we achieve this just with CSS [around here](https://github.com/liferay/liferay-portal/blob/master/modules/apps/frontend-editor/frontend-editor-ckeditor-web/src/main/resources/META-INF/resources/css/main.scss#L261-L264)? I'd like to avoid doing a patch, if we don't need to. In addition, this change makes me nervous, as maximized plugin might rely on `cke_maximized` for variety of unrelated features.

Maybe something like two different set of styles, based on combination of classes. You may use override by more specific style, or `:not()` if that's not working.

Reply:
Hello @markocikos 
I tried doing just that but unfortunately, only the editor element gets `cke_maximized` so there is no telling the stylescombo panel when the editor is maximized since the stylescombo panel element is out of scope of the editor element.

We could try adding an eventListenser for the stylecombo button in Liferay so it check if the editor is maximized and then add the css. So essentially this fix but in Liferay.

Let me know what you think.
Thank you!

Reply:
@fortunatomaldonado After looking more thoroughly into this, yeah, you are right, this cannot be fixed just with CSS. Adding a listener like you suggested could work, but it would imply moving / forking plugin to DXP, an overkill for this bugfix. We could also dynamically modify `z-index`es of toolbars, but that may have regressions elsewhere. Overall, in this case, patch like in this PR looks like the best way to go. I'd only make one or two minor changes to solution as it is:
1. We should make the new class on stylescombo something other than `cke_maximized`, so it doesn't accidentally trigger some style or code in the maximize plugin. Maybe `lfr-maximized`? 
2. Is there a native reference to set class with, something like `this.panel.className`? If it's hard to find, no problem, it's not a big deal if we keep the query selector.

Reply:
@markocikos 
1. That makes sense, I changed the class name to `lfr-maximized`
2. I wasn't able to find any reference of the panel just the contents of it. I left it as is.

Thank you for the help and let me know if there is anything else needed to change.